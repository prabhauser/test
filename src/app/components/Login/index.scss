input::-ms-clear,
input::-ms-reveal {
    display: none;
}
.login-title {
    font-size: 24px;
    line-height: 35px;
    font-weight: 600;
}
.login-left {
    min-height: 750px;
}
.login-logo {
    height: 44px;
    width: 155.48px;
    margin: 72px 0 0 55px;
}
.password-eye {
    height: 18.32px;
    width: 20px;
    margin: 0 21px 0 0px;
}
.person-icon {
    height: 24px;
    width: 18px;
    margin: 0 20px 0 19px;
}
.passwordlock-icon {
    height: 24px;
    width: 20px;
    margin: 0 20px 0 19px;
}
.password-eye:focus {
    outline: none;
}
.password-class {
    margin-top: 35px;
}
.login-snackbar {
    margin-top: 92px;
}
.login-container {
    margin: 98px 14.73% 59px 14.02%;
}
.login-heading {
    height: 36px;
    font-size: 26px;
    font-weight: 600;
    letter-spacing: 0;
    line-height: 39px;
    margin-bottom: 9px;
}
.login-headingdetail {
    height: 69px;
    font-size: 16px;
    font-weight: 500;
    letter-spacing: 0;
    line-height: 25px;
    margin-bottom: 59px;
}
.login-footer {
    margin-top: 111px;
}
.footer-copyrights {
    opacity: 0.6;
    font-size: 10px;
    font-weight: 500;
    letter-spacing: 0;
    line-height: 16px;
}
.footer-support {
    font-size: 12px;
    font-weight: 600;
    letter-spacing: 0;
    line-height: 18px;
}
.footer-contact {
    font-size: 12px;
    font-weight: 600;
    letter-spacing: 0;
    line-height: 18px;
    margin-left: -20px;
}

@media (max-width: 1200px) {
    .footer-contact {
        margin: 0;
    }
}
