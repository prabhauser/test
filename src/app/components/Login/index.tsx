import { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { userLogin, loginReset } from '../../../store/actions/loginActions';
import { connect } from 'react-redux';
import { Container, IconButton, makeStyles } from '@material-ui/core';
import VisibilityIcon from '../../../assets/images/visibility.svg';
import VisibilityOffIcon from '../../../assets/images/visibilityoff.svg';
import PersonIcon from '../../../assets/images/person.svg';
import PasswordLockIcon from '../../../assets/images/password_lock.svg';
import Logo from '../../../assets/images/dswd_logo_login.png';
import Flag from '../../../assets/images/Bitmap.png';
import Label from '../../atoms/Label/index';
import TextInput from '../../atoms/TextInput/index';
import Button from '../../atoms/Button/index';
import SnackbarAlert from '../../atoms/SnackbarAlert';
import roles from '../../constants/roles';
import { getFromLocalStorage, setToLocalStorage } from '../../../utilities/storage/storageUtility';
import YupSchemaGeneration from '../../../adb-common/validations/yup-wrapper';
import { Formik, Form, FormikProps } from 'formik';
import * as Yup from 'yup';
import { formValidations as VALIDATIONS } from '../../../adb-common/validations/rules/login';
import './index.scss';

type StateProps = {
    isLoading: boolean;
    isLoggedIn: boolean;
    loginError: any;
};

type DispatchProps = {
    userLogin: Function;
    loginReset: Function;
};

type LoginProps = StateProps & DispatchProps;

type LoginFormType = {
    userName: string;
    password: string;
};

const useStyles = makeStyles({
    loginInputAlign: {
        marginTop: '9px'
    },
    loginButtonAlign: {
        marginTop: '49px'
    }
});

//Snackbar options
const options = {
    DEFAULT: {
        message: '',
        open: false,
        type: ''
    },
    SUCCESS: {
        message: 'Updated Successfully!',
        open: true
    },
    ERROR: {
        message: 'Something went wrong.',
        type: 'error',
        open: true
    }
};

const Login = (props: LoginProps) => {
    const { isLoggedIn, loginError } = props;
    const [showPassword, setShowPassword] = useState(false);
    const [snackbarOptions, setSnackbarOptions] = useState(options.DEFAULT);
    const [disableLogin, setDisableLogin] = useState(true);
    const history = useHistory();
    const [validationSchema, setValidationSchema] = useState({});
    const [initialValues] = useState({
        userName: '',
        password: ''
    });
    const classes = useStyles();

    useEffect(() => {
        const yupSchema = VALIDATIONS?.reduce(YupSchemaGeneration, {});
        setValidationSchema(Yup.object().shape(yupSchema));
    }, []);

    useEffect(() => {
        if (isLoggedIn) {
            const roleCode = 'R01';
            setToLocalStorage('userRole', roleCode);

            const dashboardPage = Object.values(roles)?.find((element) => element?.roleCode === roleCode)?.homepage;

            if (getFromLocalStorage('loginId')) {
                history.push(`/${dashboardPage || 'dashboard'}`);
                window.scrollTo(0, 0);
            }
        }
    }, [isLoggedIn]);

    const handleClick = () => {
        setShowPassword((prev) => !prev);
    };

    // const handleSubmit = async (values: LoginFormType) => {
    //     await props.loginReset();
    //     const reqObj = {
    //         userName: values?.userName,
    //         password: values?.password,
    //         isPasswordEncrypted: false,
    //         isSkipAdAuth: true
    //     };
    //     props.userLogin(reqObj);
    // };

    //remove this code
    const handleSubmit = (values: LoginFormType) => {
        const roleCode = 'R02';
        setToLocalStorage('userRole', roleCode);
        const dashboardPage = Object.values(roles)?.find((element) => element?.roleCode === roleCode)?.homepage;
        history.push(`/${dashboardPage || 'dashboard'}`);
    };

    const handleSnackbarError = (err: any) => {
        const errorMsg = err?.displayMessage || err?.details?.alertMessage;
        const snackbarError = {
            message: errorMsg,
            type: 'error',
            open: true
        };
        setSnackbarOptions(snackbarError);
    };

    useEffect(() => {
        if (loginError) {
            handleSnackbarError(loginError);
        }
    }, [loginError]);

    const closeSnackbar = () => setSnackbarOptions(options.DEFAULT);

    return (
        <Container disableGutters component="main" maxWidth="xl">
            <SnackbarAlert className="login-snackbar" options={snackbarOptions} handleClose={closeSnackbar} />
            <div className="row mar-0">
                <div
                    className="login-left col-xl-6 col-lg-6 col-md-6 col-sm-12 col-12 pad-0"
                    style={{
                        backgroundImage: `url(${Flag})`,
                        backgroundPosition: 'center center',
                        backgroundRepeat: 'no-repeat',
                        backgroundSize: 'cover'
                    }}
                >
                    <img className="login-logo" src={Logo} />
                </div>
                <div className=" col-xl-6 col-lg-6 col-md-6 col-sm-12 col-12 pad-0">
                    <div className="d-flex flex-column ">
                        <div className=" login-container">
                            <div className="login-heading graylabel">Login</div>
                            <div className="login-headingdetail graylabel">Please login with below credentials</div>
                            <Formik
                                initialValues={initialValues}
                                validationSchema={validationSchema}
                                onSubmit={(values) => {
                                    handleSubmit(values);
                                }}
                            >
                                {(formikprops: FormikProps<LoginFormType>) => {
                                    const { values, isValid, handleChange, handleBlur, errors, touched } = formikprops;
                                    return (
                                        <Form>
                                            <div>
                                                <Label label={'User ID'} />
                                                <TextInput
                                                    variant="outlined"
                                                    fullWidth={true}
                                                    type="text"
                                                    name="userName"
                                                    value={values.userName}
                                                    autoComplete={'off'}
                                                    formInput={classes.loginInputAlign}
                                                    startAdornment={
                                                        <IconButton className="person-icon">
                                                            <img src={PersonIcon} />
                                                        </IconButton>
                                                    }
                                                    onChange={(e) => {
                                                        setDisableLogin(false);
                                                        handleChange(e);
                                                    }}
                                                    onBlur={handleBlur}
                                                    helperText={
                                                        errors.userName && touched.userName ? errors.userName : ''
                                                        //errors.userName ? errors.userName : ''
                                                    }
                                                />
                                                <Label className={'password-class'} label={'Password'} />
                                                <TextInput
                                                    variant="outlined"
                                                    fullWidth={true}
                                                    type={showPassword ? 'text' : 'password'}
                                                    name="password"
                                                    formInput={classes.loginInputAlign}
                                                    startAdornment={
                                                        <IconButton className="passwordlock-icon">
                                                            <img src={PasswordLockIcon} />
                                                        </IconButton>
                                                    }
                                                    endAdornment={
                                                        <IconButton className="password-eye" onClick={handleClick}>
                                                            {showPassword ? (
                                                                <img src={VisibilityIcon} />
                                                            ) : (
                                                                <img src={VisibilityOffIcon} />
                                                            )}
                                                        </IconButton>
                                                    }
                                                    value={values.password}
                                                    onChange={(e) => {
                                                        setDisableLogin(false);
                                                        handleChange(e);
                                                    }}
                                                    onBlur={handleBlur}
                                                    helperText={
                                                        errors.password && touched.password ? errors.password : ''
                                                        //errors.password ? errors.password : ''
                                                    }
                                                />
                                                <Button
                                                    formInput={classes.loginButtonAlign}
                                                    fullWidth={true}
                                                    name="Login"
                                                    disabled={disableLogin || !isValid}
                                                    type="submit"
                                                    variant="contained"
                                                />
                                                <div className="d-flex row no-gutters align-items-center login-footer graylabel">
                                                    <div className="col-xl-6 col-lg-6 col-md-12 col-sm-12 col-12 pad-0 footer-copyrights">
                                                        Copyright © 2021 DSWD. All rights reserved.
                                                    </div>
                                                    <div className="col-xl-4 col-lg-4 col-md-12 col-sm-12 col-12 pad-0 align-items-end footer-support">
                                                        support@dswd.com
                                                    </div>
                                                    <div className="col-xl-2 col-lg-2 col-md-2 col-sm-12 col-12 pad-0 footer-contact">
                                                        (+63)5378697580
                                                    </div>
                                                </div>
                                            </div>
                                        </Form>
                                    );
                                }}
                            </Formik>
                        </div>
                    </div>
                </div>
            </div>
        </Container>
    );
};

const mapStateToProps = (state: any) => {
    return {
        isLoggedIn: state?.login?.isLoggedIn,
        loginError: state?.login?.err,
        isLoading: state?.Test?.isLoading
    };
};

const mapDispatchToProps = {
    userLogin,
    loginReset
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
