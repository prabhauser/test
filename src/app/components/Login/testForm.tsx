import React, { useEffect, useState } from 'react';
import TextInput from '../../atoms/TextInput/index';
import ADBButton from '../../atoms/Button/index';
import YupSchemaGeneration from '../../../adb-common/validations/yup-wrapper';
import { Formik, Form, FormikProps } from 'formik';
import * as Yup from 'yup';
import { formValidations as validations } from '../../../adb-common/validations/rules/login';
import Checkbox from '../../atoms/CheckBox/index';
import RadioButton from '../../atoms/RadioButton/index';
import AutoComplete from '../../atoms/AutoComplete/index';
import Card from '../../atoms/Card/index';

interface loginFormValues {
    userName: string;
    password: string;
    subscribe: boolean;
}
// const LOCATIONS = [
//     { label: 'Ansonia, CT', value: '1', disabled: false },
//     { label: 'Avon, CT', value: '2', disabled: false },
//     { label: 'Beacon Falls, CT', value: '3', disabled: false }
// ];

const options = [
    {
        value: 1,
        label: 'test1'
    },
    {
        value: 2,
        label: 'test2'
    },
    {
        value: 3,
        label: 'test3'
    }
];

const TestForm = () => {
    const [validationSchema, setValidationSchema] = useState({});
    const [initialValues, setInitialValues] = useState({
        userName: '',
        password: '',
        subscribe: true
    });
    const [isCheckboxChecked, setIsCheckboxChecked] = useState(true);
    const [isLoading, setIsLoading] = useState(false);

    const handleSubmit = (values: loginFormValues) => {
        console.log(values);
    };
    useEffect(() => {
        const yupSchema = validations.reduce(YupSchemaGeneration, {});
        setValidationSchema(Yup.object().shape(yupSchema));
    }, []);

    const handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setIsCheckboxChecked(e.currentTarget.checked);
    };

    return (
        <div>
            Test Form Login
            <Formik
                initialValues={initialValues}
                validationSchema={validationSchema}
                onSubmit={(values) => {
                    handleSubmit(values);
                }}
            >
                {(props: FormikProps<loginFormValues>) => {
                    const { values, handleChange, errors, touched } = props;
                    return (
                        <Form>
                            <div>
                                <TextInput
                                    value={values.userName}
                                    name="userName"
                                    type="text"
                                    onChange={handleChange}
                                    helperText={errors.userName && touched.userName ? errors.userName : ''}
                                />
                                <div>
                                    <TextInput
                                        value={values.password}
                                        onChange={handleChange}
                                        name="password"
                                        type="password"
                                        helperText={errors.password && touched.password ? errors.password : ''}
                                    />
                                </div>
                                <div>
                                    <Checkbox
                                        name="subscribe"
                                        isChecked={isCheckboxChecked}
                                        value={isCheckboxChecked}
                                        onChange={(e) => {
                                            handleChange(e);
                                            handleCheckboxChange(e);
                                        }}
                                        disabled={false}
                                    />
                                </div>
                                <RadioButton disabled={false} options={['first', 'second']} label={'test'} />
                                {/* <DropDown
                                    value={values.location}
                                    items={LOCATIONS}
                                    name="location"
                                    label="location"
                                    onChange={handleChange}
                                    error={errors.location && touched.location ? errors.location : ''}
                                /> */}
                                <AutoComplete options={options} label={'Test Autocomplete'} />
                                <Card heading={'Test Card'} />
                                <ADBButton name="Login" type="submit" />
                            </div>
                        </Form>
                    );
                }}
            </Formik>
        </div>
    );
};
export default TestForm;
