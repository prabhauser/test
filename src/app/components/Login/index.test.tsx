import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Login from './index';

const mockStore = configureStore([]);

describe('Login Component', () => {
    let store;
    let component: renderer.ReactTestRenderer;
    beforeEach(() => {
        store = mockStore({
            myState: 'sample text'
        });

        component = renderer.create(
            <Provider store={store}>
                <Login />
            </Provider>
        );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});
