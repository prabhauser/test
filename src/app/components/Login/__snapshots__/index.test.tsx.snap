// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render with given state from Redux store 1`] = `
<div>
  <div>
    Test Form Login
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div>
        <div
          className="
        makeStyles-root-1
        "
        >
          <div
            className="MuiFormControl-root MuiTextField-root  makeStyles-inputWithouticon-2"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="userName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
            </div>
            
          </div>
        </div>
        <div>
          <div
            className="
        makeStyles-root-1
        "
          >
            <div
              className="MuiFormControl-root MuiTextField-root  makeStyles-inputWithouticon-2"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="password"
                  value=""
                />
              </div>
              
            </div>
          </div>
        </div>
        <div>
          <div>
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-10 Mui-checked MuiIconButton-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={true}
                    className="PrivateSwitchBase-input-12"
                    data-indeterminate={false}
                    disabled={false}
                    name="subscribe"
                    onChange={[Function]}
                    type="checkbox"
                    value={true}
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiFormControlLabel-label makeStyles-bold-7 MuiTypography-body1"
              />
            </label>
          </div>
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-13 makeStyles-primaryButton-14 undefined"
            disabled={false}
            name="Login"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Login
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
